app/scripts/modules/core/src/presentation/hooks/useEventListener.hook.spec.tsx

```diff
@@ -0,0 +1,220 @@
+  return null as JSX.Element;
+};
+
+const eventListenerOptions = { capture: true };
+
+describe('useEventListener', () => {
+  let eventTarget: HTMLDivElement;
+  let addEventListenerSpy: jasmine.Spy;
+  let removeEventListenerSpy: jasmine.Spy;
+
+  beforeEach(() => {
+    eventTarget = document.createElement('div');
+    addEventListenerSpy = spyOn(eventTarget, 'addEventListener').and.returnValue(undefined);
+    removeEventListenerSpy = spyOn(eventTarget, 'removeEventListener').and.returnValue(undefined);
+  });
+
+  it('should call addEventListener on the target element when mounted', () => {
+    mount(
+      <TestComponent element={eventTarget} eventName="keydown" listener={() => null} options={eventListenerOptions} />,
+    );
+
+    expect(addEventListenerSpy).toHaveBeenCalledTimes(1);
+    expect(removeEventListenerSpy).toHaveBeenCalledTimes(0);
+
+    expect(addEventListenerSpy.calls.argsFor(0)[0]).toBe('keydown');
+    expect(addEventListenerSpy.calls.argsFor(0)[2]).toBe(eventListenerOptions);
+  });
+
+  it('should not do anything when mounted if there is no listener prop', () => {
+    mount(<TestComponent element={eventTarget} eventName="keydown" options={eventListenerOptions} />);
+
+    expect(addEventListenerSpy).toHaveBeenCalledTimes(0);
+    expect(removeEventListenerSpy).toHaveBeenCalledTimes(0);
+  });
+
+  it('should call removeEventListener on the target element when unmounted', () => {
+    const component = mount(
+      <TestComponent element={eventTarget} eventName="keydown" listener={() => null} options={eventListenerOptions} />,
+    );
+
+    expect(addEventListenerSpy).toHaveBeenCalledTimes(1);
+    expect(removeEventListenerSpy).toHaveBeenCalledTimes(0);
+
+    component.unmount();
+
+    expect(addEventListenerSpy).toHaveBeenCalledTimes(1);
+    expect(removeEventListenerSpy).toHaveBeenCalledTimes(1);
+
+    expect(removeEventListenerSpy.calls.argsFor(0)[0]).toBe('keydown');
+    expect(removeEventListenerSpy.calls.argsFor(0)[2]).toBe(eventListenerOptions);
+  });
+
+  it('should call removeEventListener with the same event listener function reference', () => {
+    let addedListener: any;
+    let removedListener: any;
+    addEventListenerSpy.and.callFake((_: string, eventListener: () => any) => (addedListener = eventListener));
+    removeEventListenerSpy.and.callFake((_: string, eventListener: () => any) => (removedListener = eventListener));
+
+    const component = mount(
+      <TestComponent element={eventTarget} eventName="keydown" listener={() => null} options={eventListenerOptions} />,
+    );
+
+    component.unmount();
+
+    expect(addEventListenerSpy).toHaveBeenCalledTimes(1);
+    expect(removeEventListenerSpy).toHaveBeenCalledTimes(1);
+
+    expect(addedListener).toBe(removedListener);
+  });
+
+  it('should call removeEventListener with the same event listener function reference after multiple renders and new listener functions', () => {
```

Comment:
argh this was a copy-pasta/editing snafu, thanks for catching. This test file is quite long 😆 

to be totally honest, I mostly pulled in this multiple render test because of a similar one you added for `useInternalValidator` which has the same constraints around passing the same function reference to `addValidator`/`removeValidator`. I originally had this test grouped together with the one where the `listener` prop gets changed, and never fixed the title.

I am not totally confident that we need to test for multiple calls to `component.render`, changes to props seem more interesting. Might just take this test out entirely.

Reply:
took both these `it` blocks out as discussed.